ui/app/AppLayouts/Wallet/adaptors/TokenSelectorViewAdaptor.qml

```diff
@@ -0,0 +1,155 @@
+import QtQuick 2.15
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Utils 0.1
+
+import shared.stores 1.0
+
+import SortFilterProxyModel 0.2
+
+QObject {
+    id: root
+
+    /**
+      Transforms and prepares input data (assets) for TokenSelectorView needs. The assets model is internally
+      joined with `flatNetworksModel` for the `balances` submodel
+
+      Expected assets model structure:
+      - tokensKey: string -> unique string ID of the token (asset); e.g. "ETH" or contract address
+      - name: string -> user visible token name (e.g. "Ethereum")
+      - symbol: string -> user visible token symbol (e.g. "ETH")
+      - decimals: int -> number of decimal places
+      - communityId: string -> optional; ID of the community this token belongs to, if any
+      - marketDetails: var -> object containing props like `currencyPrice` for the computed values below
+      - balances: submodel -> [ chainId:int, account:string, balance:BigIntString, iconUrl:string ]
+
+      Computed values:
+      - currentCurrencyBalance: double (e.g. `1000.42` in user's fiat currency)
+      - currentBalance: double (e.g. `1.42`)
+    */
+
+    // input API
+    required property var assetsModel
+    required property var flatNetworksModel
+    required property CurrenciesStore currencyStore
+
+    // optional filter properties; empty/default values means no filtering
+    property var enabledChainIds: []
+    property string accountAddress
+    property bool showCommunityAssets
+    property string searchString
```

Comment:
Search functionality shouldn't be part of this adaptor but the component providing search bar. Search in this case is local thing, which can be done internally and don't have to be exposed.

Reply:
I see no reason to exclude it; after all it's yet another filter parameter, just like the ones above (`accountAddress` or `enabledChainIds`)

Reply:
From architectural point of view there is a reason. It's not a problem for `TokenSelectorView` which is just a list view, but the component combining that one and search bar. This component can (and should) keep search as own details, there is no need to externalize that functionality from that component. Then the component is self-contained and own the logic it offers via search text input.